src/NServiceBus.Transport.Msmq/ReceiveStrategy.cs

```diff
@@ -102,21 +101,17 @@ protected void MovePoisonMessageToErrorQueue(Message message, MessageQueueTransa
 
         protected async Task<bool> TryProcessMessage(string messageId, Dictionary<string, string> headers, Stream bodyStream, TransportTransaction transaction)
         {
-            if (discardExpiredTtbrMessages && TimeToBeReceived.HasElapsed(headers))
+            if (!ignoreIncomingTimeToBeReceivedHeaders && TimeToBeReceived.HasElapsed(headers))
             {
                 Logger.Debug($"Discarding message {messageId} due to lapsed Time To Be Received header");
                 return false;
             }
 
-            using (var tokenSource = new CancellationTokenSource())
-            {
-                var body = await ReadStream(bodyStream).ConfigureAwait(false);
-                var messageContext = new MessageContext(messageId, headers, body, transaction, tokenSource, new ContextBag());
-
-                await onMessage(messageContext).ConfigureAwait(false);
+            var body = await ReadStream(bodyStream).ConfigureAwait(false);
+            var messageContext = new MessageContext(messageId, headers, body, transaction, new ContextBag());
 
-                return tokenSource.Token.IsCancellationRequested;
-            }
+            await onMessage(messageContext).ConfigureAwait(false);
+            return false; //TODO can this be completely removed or will this be readded with CTS support?
```

Comment:
@andreasohlund with the removal of the token from the MessageContext, it looks like this doesn't need to return any bool result at all anymore. Is this intended or are there more changes coming back?

Reply:
It wasn't mentioned since MSMQ wasn't supposed to be supported in v8. That said MSMQ will only change its behaviour when running with non-transactional queues and that isn't really supported since SC can't handle it (running with non-transactional queues requires all queues in the entire system to be created as non-transactional  for things to work)

In short: Express in MSMQ never made any difference. @ramonsmits do you remember where John Breakwells blow was recovered since he had a blogpost on this topic?

Reply:
Some info here https://docs.microsoft.com/en-gb/archive/blogs/johnbreakwell/why-are-my-express-msmq-messages-being-written-to-disk but I can't find the transactional queue parts